examples/tracing-http/src/server.rs

```diff
@@ -0,0 +1,107 @@
+use hyper::{
+    service::{make_service_fn, service_fn},
+    Body, Request, Response, Server, StatusCode,
+};
+use opentelemetry::{
+    global,
+    propagation::TextMapPropagator,
+    trace::{FutureExt, Span, SpanKind, TraceContextExt, Tracer},
+    Context,
+};
+use opentelemetry_contrib::trace::propagator::trace_context_response::TraceContextResponsePropagator;
+use opentelemetry_http::{HeaderExtractor, HeaderInjector};
+use opentelemetry_sdk::{propagation::TraceContextPropagator, trace::TracerProvider};
+use opentelemetry_stdout::SpanExporter;
+use std::{convert::Infallible, net::SocketAddr};
+
+// Utility function to extract the context from the incoming request headers
+fn extract_context_from_request(req: &Request<Body>) -> Context {
+    global::get_text_map_propagator(|propagator| {
+        propagator.extract(&HeaderExtractor(req.headers()))
+    })
+}
+
+// Separate async function for the handle endpoint
+async fn handle_health_check(_req: Request<Body>) -> Result<Response<Body>, Infallible> {
+    let tracer = global::tracer("example/server");
+    let mut span = tracer
+        .span_builder("health_check")
+        .with_kind(SpanKind::Server)
+        .start(&tracer);
+    span.add_event("Health check accessed", vec![]);
+    let cx = Context::default().with_span(span);
+    let mut res = Response::new(Body::from("Server is up and running!"));
+    let response_propagator: &dyn TextMapPropagator = &TraceContextResponsePropagator::new();
```

Comment:
hint that returning context is optional, and based on the draft spec, linking to https://w3c.github.io/trace-context/#traceresponse-header